src/app/core/pricing/pricing-tier.service.ts

```diff
@@ -16,4 +18,10 @@ export class PricingTierService {
     toggleDiscount(value?: boolean): void {
         this.isVolumeDiscountActive$.next(value ?? !this.isVolumeDiscountActive$.value);
     }
+
+    updateDiscount(itemCount: number): void {
```

Comment:
where is this method called? the toggle method is not updated because the discount is not updated.

updateDiscount should be called from the cartService on updates